### PR TITLE
Enable Sonar & code improvements

### PR DESCRIPTION
**Pull request by bryan5989**
Add Sonar and Appveyor-Sonar support

### CODE REVIEW

**Comment by bryan5989** [comment]
@304NotModified I made the modifications to the run-sonar script, i think it's all set, but i can't test it. Sonar doesn't run on a pull request. Let me know if there is anything you need me to do.

**Comment by bryan5989** [comment]
@304NotModified bump

File: appveyor.yml [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
   file: '**\*.csproj'
   version: '{version}'
-  package_version: '{version}'
-  assembly_version: '{version}'
-  file_version: '{version}.{build}'
-  informational_version: '{version}'
+  package_version: '1.0.0'
+  assembly_version: '1.0'
+  file_version: '1.0.0.{build}'
+  informational_version: '1.0.0'
```
Comment (304NotModified) [inline appveyor.yml]: why is this changed? It's not that handy when releasing.

**Comment by 304NotModified** [comment]
Thanks for the changes,
I don't it get reported well

https://sonarcloud.io/dashboard?id=nlog.extensions.logging last report is from 5 aug, the build here is 11 aug

**Comment by 304NotModified** [comment]
>  Sonar: not running SonarQube, no sonar_toke

I need to check that

File: appveyor.yml (hunk 1 above)
Comment (bryan5989) [reply appveyor.yml]: I struggled with the {version} tag. {version} resolves to whatever is on line 7. So the file version was rendered {version}.{build}.{build}. And none of the numbers were making sense. I tried declaring my own variable, but couldnt get it to work.

If you look at NLog.Extensions.Logging youll notice that all the versioning stuff in dotnet_csproj (csproj patching) is ignored because you have the .\build.ps1 doing some logic to set it. I completely removed the need for the build script and just let appveyor handle it because i only have the one build target.

**Comment by bryan5989** [comment]
That aug 5 is NLog Extensions.Logging. you need a separate sonar project for this. I dont think that changes the token, but...
I at least tested that sonar runs locally on my machine and also tested pushing results with my own token to my own sonar project. And that all works. Iassumed all you had to do was make a project in sonar and everything else would stay the same. I was bot able to test if appveyor was setting variables coreectly before calling the .\run-sonar.ps1. I also noticed that by default sonar doesnt run on pull requests. Wasnt sure if that factored into this issue.

**Comment by 304NotModified** [comment]
OK thanks, will check this futher
